internal/backends/postgresql/query.go

```diff
@@ -63,9 +63,13 @@ func prepareSelectClause(schema, table string, capped, onlyRecordIDs bool) strin
 }
 
 // prepareWhereClause adds WHERE clause with given filters to the query and returns the query and arguments.
-func prepareWhereClause(p *metadata.Placeholder, sqlFilters *types.Document) (string, []any, error) {
+// It returns false if pushdown for at least one filter couldn't be applied.
+func prepareWhereClause(p *metadata.Placeholder, sqlFilters *types.Document) (string, []any, bool, error) {
 	var filters []string
 	var args []any
+	var filterPushdownAll bool
+
+	filterPushdownAll = true
```

Comment:
@noisersup do you have a suggestion of improving how to check if pushdown is applied from the entire filter?

Reply:
I didn't realize that some pushdowns with $eq and $ne were best-effort matches, causing some unexpected results. Adding a return value to differentiate between exact matches and best-effort matches in the `filterEqual` function should fix this, and added tests should cover the cases you described. 
In `filterEqual`, I followed your suggestion to set `exactMatch=false` and set it true when exact match satisfied. However, this may not work for `filterPushdownAll`, because determining whether limit pushdown can be applied is actually an AND logic, and limit pushdown can only be enabled when all fields in Filter, Sort and Skip can be applied pushdown.If `filterPushdownAll` is set to false at the beginning, then the number of conditions required to set it to true is actually uncertain, because each field in Filter, Sort and Skip needs to be checked. 
I agree that accidentally apply limit pushdown is a problem that needs to be avoided. As long as `filterpushdownall=false` in current code does not accidentally be removed during future development of new pushdowns , the worst scenario is that we fetch more results than necessary. 
Or do you @chilagrow @noisersup have any recommended ways to do this?